src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -59,24 +59,54 @@ public class HubEndPoint<THub, TClient> : IInvocationBinder where THub : Hub<TCl
 
         public async Task OnConnectedAsync(ConnectionContext connection)
         {
-            await ProcessNegotiate(connection);
+            var output = Channel.CreateUnbounded<byte[]>();
+            var connectionContext = new HubConnectionContext(output, connection);
+
+            await ProcessNegotiate(connectionContext);
```

Comment:
Actually even our client may not be able to reject SSE + binary protocol:

```CSharp
var httpConnection = new HttpConnection(new Uri(url), TransportType.ServerSentEvents);
var hubConnection = new HubConnection(httpConnection, new MessagePackHubProtocol());
```
who is supposed to reject this combination? HttpConnection has no notion of the protocol, HubConnection has no notion of transport....